src/sparseml/exporters/transforms/convert_quantizable_matmul.py

```diff
@@ -0,0 +1,315 @@
+        graph.update_node_input(
+            node=output_dequantize_node,
+            input_idx=0,
+            input_id=last_node_optional.input[0],
+        )
+
+    # create qmatmul node and add it to graph
+    qmatmul_output = (
+        matmul_node.output[0] if last_node_optional else output_quantize_node.output[0]
+    )
+    qmatmul_name = "{}_quant".format(matmul_node.name)
+    qmatmul_node = onnx.helper.make_node(
+        "QLinearMatMul",
+        qmatmul_inputs,
+        [qmatmul_output],
+        qmatmul_name,
+    )
+    matmul_node_index -= (
+        2  # adjust index to account for two deleted nodes (node_0 and node_1)
+    )
+
+    model.graph.node.insert(matmul_node_index, qmatmul_node)
+    return model
+
+
+class ConvertQuantizableMatmul(BaseTransform):
+    """
+    A transform that attempts, if appropriate, to convert MatMul nodes into
+    their quantized representation.
+    This MatMul is the result of quantizing native torch.matmul using QATMatMul
+
+    | Starting with:
+    |          INPUT_0           INPUT_1
+    |            |               |
+    |     QuantizeLinear     QuantizeLinear
+    |            |               |
+    |     DequantizeLinear   DequantizeLinear
+    |                  |      |
+    |                   MatMul
+    |                     |
+    |                  Transpose (optional)
+    |                     |
+    |                  Reshape (optional)
+    |                     |
+    |               QuantizeLinear
+    |                     |
+    |              DequantizeLinear
+    |                     |
+    |                  OUTPUT
+    |
+    | We end up converting to:
+    |          INPUT_0           INPUT_1
+    |            |               |
+    |     QuantizeLinear     QuantizeLinear
+    |                  |      |
+    |                  |      |
+    |                   QLinearMatMul
+    |                     |
+    |                   Transpose (optional)
+    |                     |
+    |                    Reshape (optional)
+    |                     |
+    |              DequantizeLinear
+    |                     |
+    |                  OUTPUT
+    """
+
+    def _transform(self, model: ModelProto) -> ModelProto:
+        graph = ONNXGraph(model)
+        count_converted_nodes = 0
+        matmul_nodes_indices, matmul_nodes = zip(
+            *[
+                (idx, node)
+                for idx, node in enumerate(model.graph.node)
+                if node.op_type == "MatMul"
+            ]
+        )
+        for idx, matmul_node in zip(matmul_nodes_indices, matmul_nodes):
+            # Check whether there is a match
+            result = is_quantizable_matmul(matmul_node, graph)
+            if result is None:
+                continue
+            output_quantize_node, last_node_optional = result
+            _LOGGER.debug(f"Matched quantizable MatMul: {matmul_node.name}")
+
+            # Convert
+            model = convert_matmul_to_quantized(
```

Comment:
@corey-nm this is the particular implementation of what you described as `transform_match` method